File: java_sdk/src/main/java/business_api_client/AdvertiserCreateBody.java
/*
 * Copyright 2023 TikTok Pte. Ltd.
 *
 * This source code is licensed under the MIT license found in
 * the LICENSE file in the root directory of this source tree.
 */

package business_api_client;

import java.util.Objects;
import java.util.Arrays;
import business_api_client.BcadvertisercreateAdvertiserInfo;
import business_api_client.BcadvertisercreateBillingGroupInfo;
import business_api_client.BcadvertisercreateBillingInfo;
import business_api_client.BcadvertisercreateContactInfo;
import business_api_client.BcadvertisercreateCustomerInfo;
import business_api_client.BcadvertisercreateQualificationInfo;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import io.swagger.v3.oas.annotations.media.Schema;
/**
 * AdvertiserCreateBody
 */

@javax.annotation.Generated(value = "com.tiktok.codegen.JavatiktokcodegenGenerator", date = "2025-03-21T10:49:20.864324-07:00[America/Los_Angeles]")
public class AdvertiserCreateBody {
  @JsonProperty("advertiser_info")
  private BcadvertisercreateAdvertiserInfo advertiserInfo = null;

  @JsonProperty("bc_id")
  private String bcId = null;

  @JsonProperty("billing_group_info")
  private BcadvertisercreateBillingGroupInfo billingGroupInfo = null;

  @JsonProperty("billing_info")
  private BcadvertisercreateBillingInfo billingInfo = null;

  @JsonProperty("contact_info")
  private BcadvertisercreateContactInfo contactInfo = null;

  @JsonProperty("customer_info")
  private BcadvertisercreateCustomerInfo customerInfo = null;

  @JsonProperty("qualification_info")
  private BcadvertisercreateQualificationInfo qualificationInfo = null;

  @JsonProperty("tied_to_billing_group")
  private Boolean tiedToBillingGroup = null;

  public AdvertiserCreateBody advertiserInfo(BcadvertisercreateAdvertiserInfo advertiserInfo) {
    this.advertiserInfo = advertiserInfo;
    return this;
  }

   /**
   * Get advertiserInfo
   * @return advertiserInfo
  **/
  @Schema(required = true, description = "")
  public BcadvertisercreateAdvertiserInfo getAdvertiserInfo() {
    return advertiserInfo;
  }

  public void setAdvertiserInfo(BcadvertisercreateAdvertiserInfo advertiserInfo) {
    this.advertiserInfo = advertiserInfo;
  }

  public AdvertiserCreateBody bcId(String bcId) {
    this.bcId = bcId;
    return this;
  }

   /**
   * Business Center ID
   * @return bcId
  **/
  @Schema(required = true, description = "Business Center ID")
  public String getBcId() {
    return bcId;
  }

  public void setBcId(String bcId) {
    this.bcId = bcId;
  }

  public AdvertiserCreateBody billingGroupInfo(BcadvertisercreateBillingGroupInfo billingGroupInfo) {
    this.billingGroupInfo = billingGroupInfo;
    return this;
  }

   /**
   * Get billingGroupInfo
   * @return billingGroupInfo
  **/
  @Schema(description = "")
  public BcadvertisercreateBillingGroupInfo getBillingGroupInfo() {
    return billingGroupInfo;
  }

  public void setBillingGroupInfo(BcadvertisercreateBillingGroupInfo billingGroupInfo) {
    this.billingGroupInfo = billingGroupInfo;
  }

  public AdvertiserCreateBody billingInfo(BcadvertisercreateBillingInfo billingInfo) {
    this.billingInfo = billingInfo;
    return this;
  }

   /**
   * Get billingInfo
   * @return billingInfo
  **/
  @Schema(description = "")
  public BcadvertisercreateBillingInfo getBillingInfo() {
    return billingInfo;
  }

  public void setBillingInfo(BcadvertisercreateBillingInfo billingInfo) {
    this.billingInfo = billingInfo;
  }

  public AdvertiserCreateBody contactInfo(BcadvertisercreateContactInfo contactInfo) {
    this.contactInfo = contactInfo;
    return this;
  }

   /**
   * Get contactInfo
   * @return contactInfo
  **/
  @Schema(description = "")
  public BcadvertisercreateContactInfo getContactInfo() {
    return contactInfo;
  }

  public void setContactInfo(BcadvertisercreateContactInfo contactInfo) {
    this.contactInfo = contactInfo;
  }

  public AdvertiserCreateBody customerInfo(BcadvertisercreateCustomerInfo customerInfo) {
    this.customerInfo = customerInfo;
    return this;
  }

   /**
   * Get customerInfo
   * @return customerInfo
  **/
  @Schema(required = true, description = "")
  public BcadvertisercreateCustomerInfo getCustomerInfo() {
    return customerInfo;
  }

  public void setCustomerInfo(BcadvertisercreateCustomerInfo customerInfo) {
    this.customerInfo = customerInfo;
  }

  public AdvertiserCreateBody qualificationInfo(BcadvertisercreateQualificationInfo qualificationInfo) {
    this.qualificationInfo = qualificationInfo;
    return this;
  }

   /**
   * Get qualificationInfo
   * @return qualificationInfo
  **/
  @Schema(description = "")
  public BcadvertisercreateQualificationInfo getQualificationInfo() {
    return qualificationInfo;
  }

  public void setQualificationInfo(BcadvertisercreateQualificationInfo qualificationInfo) {
    this.qualificationInfo = qualificationInfo;
  }

  public AdvertiserCreateBody tiedToBillingGroup(Boolean tiedToBillingGroup) {
    this.tiedToBillingGroup = tiedToBillingGroup;
    return this;
  }

   /**
   * Whether to add the ad account to a billing group. Default value: &#x60;False&#x60;. This field is an allowlist-only feature. If you would like to access it, please contact your TikTok representative
   * @return tiedToBillingGroup
  **/
  @Schema(description = "Whether to add the ad account to a billing group. Default value: `False`. This field is an allowlist-only feature. If you would like to access it, please contact your TikTok representative")
  public Boolean isTiedToBillingGroup() {
    return tiedToBillingGroup;
  }

  public void setTiedToBillingGroup(Boolean tiedToBillingGroup) {
    this.tiedToBillingGroup = tiedToBillingGroup;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    AdvertiserCreateBody advertiserCreateBody = (AdvertiserCreateBody) o;
    return Objects.equals(this.advertiserInfo, advertiserCreateBody.advertiserInfo) &&
        Objects.equals(this.bcId, advertiserCreateBody.bcId) &&
        Objects.equals(this.billingGroupInfo, advertiserCreateBody.billingGroupInfo) &&
        Objects.equals(this.billingInfo, advertiserCreateBody.billingInfo) &&
        Objects.equals(this.contactInfo, advertiserCreateBody.contactInfo) &&
        Objects.equals(this.customerInfo, advertiserCreateBody.customerInfo) &&
        Objects.equals(this.qualificationInfo, advertiserCreateBody.qualificationInfo) &&
        Objects.equals(this.tiedToBillingGroup, advertiserCreateBody.tiedToBillingGroup);
  }

  @Override
  public int hashCode() {
    return Objects.hash(advertiserInfo, bcId, billingGroupInfo, billingInfo, contactInfo, customerInfo, qualificationInfo, tiedToBillingGroup);
  }


  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class AdvertiserCreateBody {\n");
    
    sb.append("    advertiserInfo: ").append(toIndentedString(advertiserInfo)).append("\n");
    sb.append("    bcId: ").append(toIndentedString(bcId)).append("\n");
    sb.append("    billingGroupInfo: ").append(toIndentedString(billingGroupInfo)).append("\n");
    sb.append("    billingInfo: ").append(toIndentedString(billingInfo)).append("\n");
    sb.append("    contactInfo: ").append(toIndentedString(contactInfo)).append("\n");
    sb.append("    customerInfo: ").append(toIndentedString(customerInfo)).append("\n");
    sb.append("    qualificationInfo: ").append(toIndentedString(qualificationInfo)).append("\n");
    sb.append("    tiedToBillingGroup: ").append(toIndentedString(tiedToBillingGroup)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }

}

File: js_sdk/src/model/AdvertiserCreateBody.js
/*
 * Copyright 2023 TikTok Pte. Ltd.
 *
 * This source code is licensed under the MIT license found in
 * the LICENSE file in the root directory of this source tree.
 */
import {ApiClient} from '../ApiClient.js';
import {BcadvertisercreateAdvertiserInfo} from './BcadvertisercreateAdvertiserInfo.js';
import {BcadvertisercreateBillingGroupInfo} from './BcadvertisercreateBillingGroupInfo.js';
import {BcadvertisercreateBillingInfo} from './BcadvertisercreateBillingInfo.js';
import {BcadvertisercreateContactInfo} from './BcadvertisercreateContactInfo.js';
import {BcadvertisercreateCustomerInfo} from './BcadvertisercreateCustomerInfo.js';
import {BcadvertisercreateQualificationInfo} from './BcadvertisercreateQualificationInfo.js';

/**
 * The AdvertiserCreateBody model module.
 * @module model/AdvertiserCreateBody
 * @version 0.1.7
 */
export class AdvertiserCreateBody {
  /**
   * Constructs a new <code>AdvertiserCreateBody</code>.
   * @alias module:model/AdvertiserCreateBody
   * @class
   * @param advertiser_info {module:model/BcadvertisercreateAdvertiserInfo} 
   * @param bc_id {String} Business Center ID
   * @param customer_info {module:model/BcadvertisercreateCustomerInfo} 
   */
  constructor(advertiser_info, bc_id, customer_info) {
    this.advertiser_info = advertiser_info;
    this.bc_id = bc_id;
    this.customer_info = customer_info;
  }

  /**
   * Constructs a <code>AdvertiserCreateBody</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/AdvertiserCreateBody} obj Optional instance to populate.
   * @return {module:model/AdvertiserCreateBody} The populated <code>AdvertiserCreateBody</code> instance.
   */
  static constructFromObject(data, obj) {
    if (data) {
      obj = obj || new AdvertiserCreateBody();
      if (data.hasOwnProperty('advertiser_info'))
        obj.advertiser_info = BcadvertisercreateAdvertiserInfo.constructFromObject(data['advertiser_info']);
      if (data.hasOwnProperty('bc_id'))
        obj.bc_id = ApiClient.convertToType(data['bc_id'], 'String');
      if (data.hasOwnProperty('billing_group_info'))
        obj.billing_group_info = BcadvertisercreateBillingGroupInfo.constructFromObject(data['billing_group_info']);
      if (data.hasOwnProperty('billing_info'))
        obj.billing_info = BcadvertisercreateBillingInfo.constructFromObject(data['billing_info']);
      if (data.hasOwnProperty('contact_info'))
        obj.contact_info = BcadvertisercreateContactInfo.constructFromObject(data['contact_info']);
      if (data.hasOwnProperty('customer_info'))
        obj.customer_info = BcadvertisercreateCustomerInfo.constructFromObject(data['customer_info']);
      if (data.hasOwnProperty('qualification_info'))
        obj.qualification_info = BcadvertisercreateQualificationInfo.constructFromObject(data['qualification_info']);
      if (data.hasOwnProperty('tied_to_billing_group'))
        obj.tied_to_billing_group = ApiClient.convertToType(data['tied_to_billing_group'], 'Boolean');
    }
    return obj;
  }
}

/**
 * @member {module:model/BcadvertisercreateAdvertiserInfo} advertiser_info
 */
AdvertiserCreateBody.prototype.advertiser_info = undefined;

/**
 * Business Center ID
 * @member {String} bc_id
 */
AdvertiserCreateBody.prototype.bc_id = undefined;

/**
 * @member {module:model/BcadvertisercreateBillingGroupInfo} billing_group_info
 */
AdvertiserCreateBody.prototype.billing_group_info = undefined;

/**
 * @member {module:model/BcadvertisercreateBillingInfo} billing_info
 */
AdvertiserCreateBody.prototype.billing_info = undefined;

/**
 * @member {module:model/BcadvertisercreateContactInfo} contact_info
 */
AdvertiserCreateBody.prototype.contact_info = undefined;

/**
 * @member {module:model/BcadvertisercreateCustomerInfo} customer_info
 */
AdvertiserCreateBody.prototype.customer_info = undefined;

/**
 * @member {module:model/BcadvertisercreateQualificationInfo} qualification_info
 */
AdvertiserCreateBody.prototype.qualification_info = undefined;

/**
 * Whether to add the ad account to a billing group. Default value: `False`. This field is an allowlist-only feature. If you would like to access it, please contact your TikTok representative
 * @member {Boolean} tied_to_billing_group
 */
AdvertiserCreateBody.prototype.tied_to_billing_group = undefined;

